packages/url-loader/src/lib/utils.ts

```diff
@@ -0,0 +1,38 @@
+export const isArray: (data: unknown) => data is ReadonlyArray<unknown> =
```

Comment:
suggestion: Maybe it would be worth adding a JSDoc here explaining why this utility was created as at first glance it may not be understanadable and devs would need to jump here to understand why normal `isArray` function was not used.

Reply:
Makes sense, I'll add that. `Array.isArray` sucks for a couple reasons:

- Narrows the type to `any[]`
- Doesn't work with `readonly` arrays

Reply:
thats interesting - can't help to wonder if there's any kinda native workaround / future spec that'll address that kinda thing. perhaps it wasn't the main intent, but i used to always use Object.keys but i learned that Object.entries is better for TS. wonder if there's an equivalent either now or in the future? but maybe there's not much additional value that could be provided so its not worth the effort?

Reply:
`Array.isArray` would be much safer if it was defined like the version I implemented, but at this point it would be too big of a breaking change for TS since so much code is implicitly relying on that `any` ☠️

Luckily, there's no runtime overhead here as it's just a redeclared type with a reference to the original method.

`Object.keys` and `Object.entries` are a bit different. In those cases, TS has opted for the safe option by inferring a conservative type, but that isn't necessarily what you want if you have an object with known keys and want the narrowed types. That's what the `entriesOf` utility I added can help with.

Reply:
cool thanks david